### PR TITLE
Updates to Templates #11

### PR DESCRIPTION
**Pull request by jgclark**
Structure and readability changes only, I hope.

### CODE REVIEW

**Comment by jgclark** [comment]
@dwertheimer may want to pitch in with ideas too?

File: nmn.Templates/src/interpolation.js [hunk 1]
```diff
@@ -50,18 +49,44 @@ async function processTag(
   return await getInput(`Value for ${tag}`)
 }
 
+// Apply any matching tag functions
+export async function processTags(
```
Comment (nmn) [inline nmn.Templates/src/interpolation.js:53]: This function is defined twice

**Review by nmn — CHANGES_REQUESTED** [review]

File: nmn.Templates/src/interpolation.js (hunk 1 above)
Comment (dwertheimer) [reply nmn.Templates/src/interpolation.js:53]: Exists in `interpolation.js` and `tagFunctions.js`

**Comment by jgclark** [comment]
Fixing multiple definition: going back to a single file 'interpolation.js' file, though I don't think the name is very clear. I was trying to make a smaller file where it was clearer that people should add new functions, but circular defintions stopped that.

(PS I'm finding my way around the GitHub extensions in VSC quite confusing for PRs, so apologies if I goof in the process.)

File: nmn.Templates/src/interpolation.js (hunk 1 above)
Comment (jgclark) [reply nmn.Templates/src/interpolation.js:33]: Thanks. Hopefully the next commit sorts this.

**Comment by jgclark** [comment]
Not having this resolved was causing me increasing pain as it was blocking getting other code released. So, as I've not had any comments in 5 days, I've gone ahead and made the commits. 

The main changes are:
- moving my weather lookup code from DailyJournal to Templates plugin
- making it more obvious where to add new functions to Templates
- renaming 'addTemplate' function to 'applyTemplate'.
- adding 'applyNamedTemplate' function

**Pull request merged**